smack-extensions/src/main/java/org/jivesoftware/smackx/pubsub/PubSubManager.java

```diff
@@ -307,7 +307,8 @@ private LeafNode getOrCreateLeafNodeProsodyWorkaround(final String id)
             return createNode(id);
         }
         catch (XMPPErrorException e1) {
-            if (e1.getXMPPError().getCondition() == Condition.conflict) {
+            if (e1.getXMPPError().getCondition() == Condition.conflict ||
+                    e1.getXMPPError().getCondition() == Condition.service_unavailable) {
```

Comment:
I tested it and it gives the service-unavailable error only :/
I don't understand, why we retry to create the node in the workaround method anyway. I mean, we tried before in the method above and got to the workaround method because it failed, right?